app/operations/beta/project_board_operations/read.rb

```diff
@@ -0,0 +1,66 @@
+module Beta
+  module ProjectBoardOperations
+    class Read
+      include Operation
+
+      def initialize(project_id:, current_user:, current_flow: nil, projects_scope: Project)
+        @project_id = project_id
+        @current_user = current_user
+        @current_flow = current_flow
+        @projects_scope = projects_scope
+      end
+
+      def call
+        project_board = yield create_project_board
+
+        Success(project_board)
+      rescue ActiveRecord::RecordNotFound => e
+        Failure(e)
+      end
+
+      private
+
+      attr_reader :project_id, :current_user, :current_flow, :projects_scope
+
+      def create_project_board
+        project_users = project.users
+
+        project_board_params = stories_and_past_iterations.merge(
+          project: project,
+          users: project_users,
+          current_user: current_user,
+          current_flow: current_flow,
+          default_flow: default_flow,
+          labels: project_labels
+        )
+
+        project_board = ::ProjectBoard.new(project_board_params)
+
+        Success(project_board)
+      end
+
+      def stories_and_past_iterations
+        read_all_result = yield ::StoryOperations::ReadAll.call(project: project)
+        read_all_result[:stories] = read_all_result.delete(:active_stories)
+        read_all_result
+      end
+
+      def project_labels
+        labels = project.stories.map(&:labels).reject { |label| label.to_s.empty? }
+        labels.join(',').split(',').uniq.join(',')
```

Comment:
Why are you `join` and then `split` here?

It seems you don't need that 🤔 

```diff
- labels.join(',').split(',').uniq.join(',')
+ labels.uniq.join(',')
```

Another thing, you may write like this:

```ruby
def project_labels
  project.stories
         .map(&:labels)
         .reject { |label| label.to_s.empty? }
         .uniq
         .join(',')
end
```

Reply:
actually Idk. didn't change the previous code, but I'll try

Reply:
I see the problem now.
each entry of label can be a set of labels, so the `.map` return something like `["front,back", "front", "bug,back"]`
that's why it is using the "double" `join`.

Reply:
changed variable names to address this